src/main/resources/assets/js/components/AppHealthComponent.jsx

```diff
@@ -0,0 +1,43 @@
+/** @jsx React.DOM */
+
+"use strict";
+
+var React = require("react/addons");
+
+module.exports = React.createClass({
+    propTypes: {
+      data : React.PropTypes.array.isRequired,
+      className : React.PropTypes.string
+    },
+
+    getDefaultProps: function () {
+      return {
+      };
+    },
+
+    render: function () {
+      // normalize data to add up to 100%
+      var dataSum = this.props.data.reduce(function (a, x) {
+        return a + x.quantity;
+      }, 0);
+      var normalizedData = this.props.data.map(function (d) {
+        return {
+          quantity: d.quantity * 100 / dataSum,
+          className: ("progress-bar " + (d.className || "")).trim()
```

Comment:
The current implementation is trivial IMO. Why use anything more advanced than reduce + map?